src/viur/core/modules/user.py

```diff
@@ -1138,22 +1140,41 @@ class User(List):
     verifyEmailAddressMail = "user_verify_address"
     passwordRecoveryMail = "user_password_recovery"
 
-    authenticationProviders: list[UserAuthentication] = [
+    authenticationProviders: t.Iterable[UserPrimaryAuthentication] = (
         UserPassword,
-        GoogleAccount
-    ]
+        conf.user.google_client_id and GoogleAccount,
+    )
```

Comment:
I find it inappropriate to allow None here. There should only be real classes here, as required by the type hint. Checking this every time at runtime in the methods with an if is on the one hand unnecessary duplicate code but also avoidable performance losses.

Simply filter out the values:

```py
    authenticationProviders: t.Iterable[UserPrimaryAuthentication] = tuple(filter(None(
        UserPassword,
        conf.user.google_client_id and GoogleAccount,
    ))
```